server/channels/app/content_flagging_test.go

```diff
@@ -12,88 +12,323 @@ import (
 		require.True(t, status)
 
-		config = getBaseConfig()
-		config.ContentFlaggingSettings.ReviewerSettings.TeamAdminsAsReviewers = model.NewPointer(true)
-		config.ContentFlaggingSettings.ReviewerSettings.SystemAdminsAsReviewers = model.NewPointer(true)
+		config.ReviewerSettings.TeamAdminsAsReviewers = model.NewPointer(true)
+		config.ReviewerSettings.SystemAdminsAsReviewers = model.NewPointer(true)
+		appErr = th.App.SaveContentFlaggingConfig(config)
+		require.Nil(t, appErr)
 
-		status = ContentFlaggingEnabledForTeam(config, "team1")
+		status, appErr = th.App.ContentFlaggingEnabledForTeam("team1")
+		require.Nil(t, appErr)
 		require.True(t, status)
 	})
 }
 
+func TestSaveContentFlaggingConfig(t *testing.T) {
+	mainHelper.Parallel(t)
+	th := Setup(t).InitBasic()
+	defer th.TearDown()
+
+	th.App.Srv().SetLicense(model.NewTestLicenseSKU(model.LicenseShortSkuEnterpriseAdvanced))
+
+	t.Run("should save content flagging config successfully", func(t *testing.T) {
+		config := model.ContentFlaggingSettingsRequest{
+			ContentFlaggingSettingsBase: model.ContentFlaggingSettingsBase{
+				EnableContentFlagging: model.NewPointer(true),
+				AdditionalSettings: &model.AdditionalContentFlaggingSettings{
+					ReporterCommentRequired: model.NewPointer(true),
+					HideFlaggedContent:      model.NewPointer(false),
+					Reasons:                 &[]string{"spam", "harassment", "inappropriate"},
+				},
+			},
+			ReviewerSettings: &model.ReviewSettingsRequest{
+				ReviewerSettings: model.ReviewerSettings{
+					CommonReviewers:         model.NewPointer(true),
+					SystemAdminsAsReviewers: model.NewPointer(true),
+					TeamAdminsAsReviewers:   model.NewPointer(false),
+				},
+				ReviewerIDsSettings: model.ReviewerIDsSettings{
+					CommonReviewerIds: []string{th.BasicUser.Id, th.BasicUser2.Id},
+				},
+			},
+		}
+		config.SetDefaults()
+
+		appErr := th.App.SaveContentFlaggingConfig(config)
+		require.Nil(t, appErr)
+
+		// Verify system config was updated
+		savedConfig := th.App.Config()
+		require.Equal(t, *config.EnableContentFlagging, *savedConfig.ContentFlaggingSettings.EnableContentFlagging)
+		require.Equal(t, *config.ReviewerSettings.CommonReviewers, *savedConfig.ContentFlaggingSettings.ReviewerSettings.CommonReviewers)
+		require.Equal(t, *config.ReviewerSettings.SystemAdminsAsReviewers, *savedConfig.ContentFlaggingSettings.ReviewerSettings.SystemAdminsAsReviewers)
+		require.Equal(t, *config.ReviewerSettings.TeamAdminsAsReviewers, *savedConfig.ContentFlaggingSettings.ReviewerSettings.TeamAdminsAsReviewers)
+		require.Equal(t, *config.AdditionalSettings.ReporterCommentRequired, *savedConfig.ContentFlaggingSettings.AdditionalSettings.ReporterCommentRequired)
+		require.Equal(t, *config.AdditionalSettings.HideFlaggedContent, *savedConfig.ContentFlaggingSettings.AdditionalSettings.HideFlaggedContent)
+		require.Equal(t, *config.AdditionalSettings.Reasons, *savedConfig.ContentFlaggingSettings.AdditionalSettings.Reasons)
```

Comment:
You don't need to assert settings individually `require.Equal` can actually compare structs (even if they are pointer values)

Reply:
They are two different structs - `config` is `*model.Config` and `savedConfig` is `*model.ContentFlaggingSettingsRequest`. It doesn't work with different structs.